### PR TITLE
Move Identifier hashing into Self's Impl

### PR DESCRIPTION
**Pull request by madig**

### CODE REVIEW

File: src/shared_types.rs [hunk 1]
```diff
@@ -76,6 +78,13 @@ impl Identifier {
     pub fn as_str(&self) -> &str {
         &self.0
     }
+
+    /// Return the u64 hash value of self.
+    pub fn hash_self(&self) -> u64 {
```
Comment (cmyr) [inline src/shared_types.rs:83]: I might just call this method `hash`?

**Review by cmyr — APPROVED** [review]
Looks good! `hash_self` feels a bit funky as a method name to me, and I think just `hash` should be fine?

**Comment by madig** [comment]
Wait, I'm not done yet! Haven't gotten around to the actual splitting out part.

File: src/shared_types.rs (hunk 1 above)
Comment (madig) [reply src/shared_types.rs:83]: I think this is already taken by the Hash trait :D

**Comment by cmyr** [comment]
hahaha but this is such a nice little PR, so easy to merge!

File: src/shared_types.rs (hunk 1 above)
Comment (cmyr) [reply src/shared_types.rs:83]: the same name can exist on a type and a trait, that's no problem.

Comment (cmyr) [reply src/shared_types.rs:83]: ... I think?

Comment (madig) [reply src/shared_types.rs:83]: Only one  way to find out

Comment (madig) [reply src/shared_types.rs:83]: Hm, nope. It would result in a recursive call, see the body of the function.

Comment (cmyr) [reply src/shared_types.rs:83]: try:

```rust
    pub fn hash(&self) -> u64 {
        let mut hasher = DefaultHasher::new();
        Hash::hash(self, &mut hasher);
        hasher.finish()
    }
```

**Pull request merged**